tuned/plugins/plugin_disk.py

```diff
@@ -27,22 +27,38 @@ def __init__(self, *args, **kwargs):
 	def _init_devices(self):
 		super(DiskPlugin, self)._init_devices()
 		self._devices_supported = True
+		self._use_hdparm = True
 		self._free_devices = set()
+		self._hdparm_devices = set()
 		for device in self._hardware_inventory.get_devices("block"):
 			if self._device_is_supported(device):
 				self._free_devices.add(device.sys_name)
+				if self._use_hdparm and self._is_hdparm_supported(device.sys_name):
+					self._hdparm_devices.add(device.sys_name)
 
 		self._assigned_devices = set()
 
 	def _get_device_objects(self, devices):
 		return [self._hardware_inventory.get_device("block", x) for x in devices]
 
+	def _is_hdparm_supported(self, device):
+		(rc, out, err_msg) = self._cmd.execute(["hdparm", "-C", "/dev/%s" % device], \
+				no_errors = [errno.ENOENT], return_err=True)
+		if rc == -errno.ENOENT:
+			log.warn("hdparm command not found, ignoring for other devices")
+			self._use_hdparm = False
+			return False
+		elif rc:
+			log.debug("device %s not supported by hdparm (rc: %s, msg: '%s')" % (device, rc, err_msg))
+			return False
+		return "unknown" not in out
```

Comment:
I think you should also log debug it, e.g. hdparm returned uknown for the drive, or similarly.

Reply:
I meant the '"unknown" in out' condition.